### ISSUE TITLE
Don't do duplicate work

### ISSUE THREAD

**Issue by ENT8R** [Teester/Whats-Nearby#6]
There is already an existing application (https://github.com/westnordost/StreetComplete), I'm sure you are aware of, which has nearly the same purpose of this app too (showing easy-to-answer-questions to users which don't know much about OSM)
Is there a special reason why you created such a similar app? I mean one obvious reason might be the restriction to show only questions about places the user actually visited... But SC asks the user before he answers a question for a place he might not be at whether he is really at this place.
So SC is not as strict as your app, but I think @westnordost wanted to add a bit more restriction in the future...

**Comment by westnordost** [comment]
Oh, didn't know about this project. I read a bit into the description and the source and to what I understand, the concept seems to be pretty much the same as SC, which you no doubt know, since you already contributed to it.
The existence of this project kind of makes me sad, because it means that while you liked the concept of SC, there was something in that project that was so wrong to the point where you decided that it will not be possible to fix that and create your own app instead of pulling together.

**Comment by Teester** [comment]
Please don't consider the existence of _What's Nearby?_ as a criticism of StreetComplete.  There are some fundamental differences in the concepts behind both apps.  StreetComplete is a fantastic project, and I'm certainly not trying to make a copy of it with this project.  Indeed, without @westnordost's marvelous [osmapi library](https://github.com/westnordost/osmapi) it would have taken far longer to get as far as it has.

Admittedly, the idea for _What's Nearby?_ came partly from a [StreetComplete issue](https://github.com/westnordost/StreetComplete/issues/602).  The issue deals with the automatic monitoring of the users location to prompt them about nearby quests.  Other inspirations were found in Foursquare and Google Maps apps, both of which include features where they ask questions about locations users have visited via notifications.  I had been a big fan of these features and had previously wondered if something similar would be possible for OSM.  An evening of playing with overpass turbo to produce queries like [this](http://overpass-turbo.eu/s/rA1) led me to believe that it was doable.

_What's Nearby?_ wants you to help the map when you don't have time to map.  To this end, it's specifically designed to get your attention for 10 or 15 seconds at a time and then disappear.  StreetComplete, on the other hand, is designed to entice you to map.  All those quests close by just beg to be solved.  The apps serve different needs and there's no reason why they can't coexist, even on the same device.  While certain concepts (user friendly questions) are common to both apps, I'm not sure that the functionality of _What's Nearby?_ could easily be integrated into StreetComplete.

I hope you understand where I'm coming from.  I believe that a rich ecosystem of apps that help people contribute in different ways is vital for the continuing success of OSM.  While _What's Nearby?_ might not end up having the same impact as editors like iD, JOSM or even StreetComplete, I believe that any contributions that it produces which might otherwise go uncontributed can only be good for the project.

**Comment by westnordost** [comment]
Ah, I see, so it is a slightly different niche.

**Comment by zossebart** [comment]
And it does it's job very well! I use Streetcomplete often in situations where i have some minutes of time (e.g. waiting for someone) to quickly add some details to the map, whereas Whats-Neatby is nice for situations where you don't have time! I like both tools very much!

**Comment by ENT8R** [comment]
Thank you for your detailed response! I think that this issue can be closed now.

**Comment by rugk** [comment]
Uhh, just found that issue (and app) and I really don't think this is not on a good way here... I really appreciate your work, @Teester, you've put into this, and (somewhat) especially because of that, I would avoid duplicate work here.

>What's Nearby? wants you to help the map when you don't have time to map.

At least for me, I use StreetComplete in exactly the same way. At least SC is of course no full editor like Vespuccci or so, and this app seems to do the exact same thing: Ask questions about OSM. How much time I spend on that is a thing I can easily control by myself.

> I'm not sure that the functionality of What's Nearby? could easily be integrated into StreetComplete.

Why?
I guess that's the main point here, so let's look at the functionaility and compare it.

Whats-Naerby only:
* notifications/display of quests in the forgeground or so (only looked at screenshots)
* queries automatically every x minutes

SC only:
* map
* quests displayed on map

But, IMHO, the main thing this is all about, i.e. the quests and the whole querying stuff, how the quests re displayed/how they are asked/what UI to use/etc, is common to both apps.
As such, not unifying it is quite a loss, as I do not see any reason why one quest may be in one app, but not in another one. Both apps try to make "fast answer" quests.

And seeing your TODO with "Add notes to the map" this really seems like duplicating SC functionality if you also want to icnlude a map, later...

So, basically, I'd say you duplicate the most difficult thing here, which is making quests that are good and easy to use. 
And of course it is possible and relatively easy (compared to writing another app) to add a query in the background to show you some quests in the notifications. Actually, this is an [often](https://github.com/westnordost/StreetComplete/issues/48) [requested](https://github.com/westnordost/StreetComplete/issues/48 [feature](https://github.com/westnordost/StreetComplete/issues/1193), so maybe it would even make sense (and one issue is not even closed/dismissed yet of the ones, I've linked).
